frcxx_nft/src/state.rs

```diff
@@ -17,25 +19,12 @@ use thiserror::Error;
 
 pub type TokenID = u64;
 
-#[derive(Error, Debug)]
-pub enum StateError {
-    #[error("ipld amt error: {0}")]
-    IpldAmt(#[from] AmtError),
-    #[error("ipld hamt error: {0}")]
-    IpldHamt(#[from] HamtError),
-    #[error("token id not found: {0}")]
-    TokenNotFound(TokenID),
-    /// This error is returned for errors that should never happen
-    #[error("invariant failed: {0}")]
-    InvariantFailed(String),
-}
-
 /// Each token stores its owner, approved operators etc.
 #[derive(Serialize_tuple, Deserialize_tuple, Clone, Debug)]
 pub struct TokenData {
     pub owner: ActorID,
     // operators on this token
-    pub operators: Vec<ActorID>, // or maybe as a Cid to an Amt
+    pub operators: HashSet<ActorID>, // or maybe as a Cid to an Amt
     pub metadata_id: Cid,
```

Comment:
I'm new to NFT standards so I'm guessing I'm the one who is confused, but I would expect this to be `data_id`.

Reply:
Are you referring to `pub owner: ActorID`?

`TokenData` is used in `Amt<TokenData>` where the index of the Amt is implicitly the TokenID 

So we need to store the owner, operators, metadata of the NFT in this structure

Reply:
ah sorry no I'm referring to `metadata_id`. My naive understanding is that NFTs can include arbitrary data which this field accounts for and it is a bit strange to me that it's called metadata given that I think of the data as the main event.  However like I said this is all new to me so feel free to ignore.